polaris-core/src/main/java/org/apache/polaris/core/persistence/PolarisMetaStoreManagerImpl.java

```diff
@@ -698,12 +703,14 @@ private void bootstrapPolarisService(
 
   /** {@inheritDoc} */
   @Override
-  public @NotNull BaseResult bootstrapPolarisService(@NotNull PolarisCallContext callCtx) {
+  public @NotNull BaseResult bootstrapPolarisService(
+      @NotNull PolarisCallContext callCtx, PrincipalSecretsGenerator rootSecretsGenerator) {
```

Comment:
Rather than changing the public interface, can we make the `PrincipalSecretsGenerator` a constructor param? I don't think callers need to know anything about this

Reply:
Different generators are used during bootstrapping and (REST) API-driven principal creation.

Reply:
reworked to avoid modifying public interfaces.